### DISCUSSION TITLE
Vary number of columns with window size

### DISCUSSION THREAD

**Discussion by thom-cameron** [sxyazi/yazi#1677] [2. Show and tell]
In small windows, maintaining the default 3 column layout and ratios makes each column too narrow to be useful. Just wanted to share my solution for varying the number of columns displayed based on the size of the window as is available in [gokcehan's lf](https://github.com/gokcehan/lf/wiki/Tips#dynamically-set-number-of-columns). 

Here's the patched `Tab:layout()` I've put in my init.lua:

``` lua
--- change number of columns in small windows
function Tab:layout()
    if self._area.w > 80 then
        self._chunks =
            ui.Layout():direction(ui.Layout.HORIZONTAL):constraints(
            {
                ui.Constraint.Ratio(MANAGER.ratio.parent, MANAGER.ratio.all),
                ui.Constraint.Ratio(MANAGER.ratio.current, MANAGER.ratio.all),
                ui.Constraint.Ratio(MANAGER.ratio.preview, MANAGER.ratio.all)
            }
        ):split(self._area)
    else
        self._chunks =
            ui.Layout():direction(ui.Layout.HORIZONTAL):constraints(
            {
                ui.Constraint.Ratio(0, MANAGER.ratio.all),
                ui.Constraint.Ratio(MANAGER.ratio.current + MANAGER.ratio.parent, MANAGER.ratio.all),
                ui.Constraint.Ratio(MANAGER.ratio.preview + MANAGER.ratio.parent, MANAGER.ratio.all)
            }
        ):split(self._area)
    end
end
```

**Comment by aShabat** [comment]
Hi, where did you find documentation about fields like _area and _chunks?

**Reply by thom-cameron** [reply]
I didn't, sorry. I just searched around in the codebase

**Reply by rieje** [reply]
There's [toggle-pane.view](https://github.com/yazi-rs/plugins/tree/main/toggle-pane.yazi) now for something more predictable and versatile but manual.
